### ISSUE TITLE
Data source for newrelic_entity returning incorrect ID

### ISSUE THREAD

**Issue by pickgr** [newrelic/terraform-provider-newrelic#1041]
### Terraform Version
Terraform v0.12.29
+ provider.newrelic v2.13.5

### Affected Resource(s)
Please list the resources as a list, for example:
- `newrelic_alert_condition`

### Terraform Configuration
> Please include your `provider` configuration (sensitive details redacted) as well as the configuration of the resources and/or data sources related to the bug report.
```hcl
terraform {
  required_version = "= 0.12.29"
}

provider "newrelic" {
  account_id    = var.account_id
  admin_api_key = var.admin_api_key
  api_key       = var.api_key
  region        = var.region
  version       = "= 2.13.5"
}

data "newrelic_entity" "service1" {
  name = "Service 1"
}

data "newrelic_entity" "service2" {
  name = "Service 2"
}

resource "newrelic_alert_condition" "apdex_critical_service_apps" {
  condition_scope = "application"

  entities = [
    data.newrelic_entity.service1_id,
    data.newrelic_entity.service2_id
  ]

  metric = "apdex"
  name   = "Service Apdex (Low) - Critical"

  policy_id = data.newrelic_alert_policy.foo.id

  term {
    duration      = 10
    operator      = "below"
    priority      = "critical"
    threshold     = "0.5"
    time_function = "all"
  }

  type = "apm_app_metric"
}
```

### Actual Behavior
The data source for data.newrelic_entity.service2_id is returning an incorrect ID.  service1 works OK

```
Error: 422 response returned: Invalid entity id(s): 123456790.
```

### Expected Behavior
All data sources should return the correct ID.

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform init`
2. `terraform plan` # shows drift for the ID in the entiries
3. `terraform apply` # shows the error above

### Important Factoids
I suspect this is account specific and will be difficult to reproduce without access to our account.  Also, this problem is not see when using the older data source for `newrelic_application`. Upgrading to use `newrelic_application` caused this issue.

We changed the data source because of this warning:

```
Warning: Use the `newrelic_entity` data source instead.
```

### References
Are there any other GitHub issues (open or closed) or Pull Requests that should be linked here? For example:
Could not find anything

**Comment by jpvajda** [comment]
I'll try to reproduce this today thanks for making us aware of the problem @pickgr

**Comment by jpvajda** [comment]
@pickgr curious are these entity names 

```  
entities = [
    data.newrelic_entity.service1_id,
    data.newrelic_entity.service2_id
  ] ```

obfuscated? I was looking up the entity on our side and couldn't find any by this name associated with your account.

**Comment by pickgr** [comment]
@jpvajda those were dummy names I used for the PR.

These are the actual

-    data.newrelic_entity.firestorm_service_prod_useast.application_id
+    # TODO - the following returns 705936293 ??
+    # data.newrelic_entity.firestorm_service_prod_useast.application_id

**Comment by pickgr** [comment]
`# TODO
# data "newrelic_entity" "firestorm_service_prod_useast" {
data "newrelic_application" "firestorm_service_prod_useast" {
  name = "Firestorm Service - Prod - IBM - USEast"
}`

**Comment by jpvajda** [comment]
@pickgr Thank you! I do see 2 existing alerts on that entity, 1 seems to match your Terraform configuration . Policy Id: 679620

```
  | Firestorm Service Service Apdex (Low) - Critical | Apdex   <   0.5    for at least   10m    CRITICAL | Sep 11, 8:04 am | View policy
```

Was this an attempt to create a new alert condition on that entity or modify an existing?

**Comment by pickgr** [comment]
@jpvajda we were trying to modify an existing and replace the `newrelic_application` data source which is being deprecated per the warning above with `newrelic_entity`

**Comment by ctrombley** [comment]
Hey @pickgr!  You'll want to use `data.newrelic_entity.service1.application_id` instead of `data.newrelic_entity.service1.id` when populating the `entities` field in `newrelic_alert_condition`.  It is a bit of an unfortunate naming convention that was put in place a while ago that has led to some confusion. Give that a try and let us know how it goes!

**Comment by pickgr** [comment]
@ctrombley please see my recent comments above.  We are using `data.newrelic_entity.service1.application_id` and it returns a different ID from when we used `data.newrelic_application.service.id`

We have a meeting setup with our account team on (Brian Rogers) Monday to hopefully demonstrate the issue so it's clear (

**Comment by zlesnr** [comment]
At a quick glance, this looks similar to https://github.com/newrelic/terraform-provider-newrelic/issues/1016
In some cases the API returns a different ID than the user is expecting, and so in the case of #1016 because it was a browser entity, the reference needed to point at the `browser_id` iirc.

**Comment by ctrombley** [comment]
Hey @pickgr, I think I see what's happening here.  You have an app named "Firestorm Service - Prod - IBM - USEast" in two of your accounts, but the `newrelic_entity` data source is not yet defaulting its `account_id` from the provider config.  The API is returning 2 applications and the provider is just grabbing the first one, which in this case is incorrect.  We'll take an action item to get that in place, but for the meantime you should be able to specify your account ID in the `newrelic_entity` data source's config with the `accountId` attribute to uniquely identify your application.

**Comment by pickgr** [comment]
@ctrombley - that definitely sounds like it could be the issue.  Thanks!

I'm having trouble with the work-around you suggested.

```
Error: Unsupported argument

  on data.tf line 47, in data "newrelic_entity" "firestorm_service_prod_useast":
  47:   accountId = "${var.account_id}"

An argument named "accountId" is not expected here. Did you mean "account_id"?
```

So I tried `account_id` and get this:

Error: "account_id": this field cannot be set

```
  on data.tf line 46, in data "newrelic_entity" "firestorm_service_prod_useast":
  46: data "newrelic_entity" "firestorm_service_prod_useast" {
```

Here's the code:

```
data "newrelic_entity" "firestorm_service_prod_useast" {
  account_id = "${var.account_id}"
  name       = "Firestorm Service - Prod - IBM - USEast"
}
```

**Comment by jpvajda** [comment]
@pickgr try  `account_id = var.account_id` 

I have a test setup here that you can reference: https://github.com/jpvajda/terraform-project/tree/main/newrelic

**Comment by ctrombley** [comment]
@pickgr I made a mistake here, it looks like we don't have `account_id` plumbed in as a search parameter like I thought we did.  It's just a computed attribute.  However, what you can do is use the automatic `accountId` entity tag in your data source config:

```
data "newrelic_application" "firestorm_service_prod_useast" {
  name = "Firestorm Service - Prod - IBM - USEast"
  tag {
    key = "accountId"
    value = "${var.account_id}"
  }
}
```

Let us know if this works!

**Comment by pickgr** [comment]
@ctrombley thank you!  Specifying the accountID via the tag in the data source does indeed work-around the issue.  Looking forward to the official fix.

**Comment by pavel-kurnosov** [comment]
I think the problem still persists. If I have in two subscriptions app with the same name, the provider picks randomly and fail with an incorrect ID. The solution with the tag is working, but I don't think it is by design. Entity should be picked from correct account automatically.

**Comment by neubarth** [comment]
We also have this issue. The workaround with the `tag` works (also for "newrelic_entity").

**Comment by kidk** [comment]
This is expected behaviour as the entity search is account agnostic. I agree this is confusing so I'd like to explore either improving the docs, or adding additional functionality to the resource.

**Comment by alecxvs** [comment]
Filtering the accountid tag is a good solution to the problem but it's not obvious enough. I just ran into this and it wasn't failing but causing unexpected modifications to one of my apps. At the very least, the data source documentation needs to be further updated to make it very clear that this will search all accounts that your api key user has access to, and can cause unexpected results if the name matches in multiple accounts that your api key user has access to. Big red box type of thing. 

There is an example of using the account id tag currently but it's in the middle of a lengthy example so it's too easy to miss when this can cause unexpected behaviors.

I wouldn't mind the default being changed to query a single account with a boolean to search all accounts, though given that I have duplicates across accounts, that wouldn't be useful for me since I can't get all of those apps with the same name.

**Comment by alecxvs** [comment]
Actually, does the provider even use the account_id? I'm currently using newrelic_account to get the id based on the account name provided in config; it has no connection to the account id passed to the provider. We could leverage this to use only one api key for all of our accounts which would save me a lot of time switching keys when I'm deploying changes for several accounts. Whether or not that's good or secure I do not know, but new relic's apis apparently allow it.

**Comment by pranav-new-relic** [comment]
Hi there, a fix to this issue has been released with [v3.24.1](https://github.com/newrelic/terraform-provider-newrelic/releases/tag/v3.24.1) of the Terraform Provider. Closing this issue as the fix is out - we shall plan on adding a note to the documentation for users to be watchful while using this resource, when there are entities across subaccounts, with the same name. Thank you for reporting this.
